Highlight/Highlighter.php

```diff
@@ -70,56 +70,52 @@ public function __construct()
     private function registerLanguages()
     {
         // Languages that take precedence in the classMap array.
-        foreach (array("xml", "django", "javascript", "matlab", "cpp") as $l) {
-            $this->createLanguage($l);
+        $languagePath = __DIR__ . DIRECTORY_SEPARATOR . "languages" . DIRECTORY_SEPARATOR;
+        foreach (array("xml", "django", "javascript", "matlab", "cpp") as $languageId) {
+            $filePath = $languagePath . $languageId . ".json";
+            if (is_readable($filePath)) {
+                $this->registerLanguage($languageId, $filePath);
```

Comment:
While `$this->` does work fine here, `registerLanguage()` is a static function so it should be `self::registerLanguage()`.